aesm-client/src/imp/sgx.rs

```diff
@@ -0,0 +1,119 @@
+use std::time::Duration;
+use std::net::TcpStream;
+use std::io::{Read,Write};
+
+use byteorder::{LittleEndian, NativeEndian, ReadBytesExt, WriteBytesExt};
+use protobuf::Message;
+
+pub use error::{AesmError, Error, Result};
+use {
+    AesmRequest, FromResponse, QuoteInfo, QuoteResult, QuoteType,
+    Request_GetQuoteRequest, Request_InitQuoteRequest,
+};
+
+/// This timeout is an argument in AESM request protobufs.
+///
+/// This value should be used for requests that can be completed locally, i.e.
+/// without network interaction.
+const LOCAL_AESM_TIMEOUT_US: u32 = 1_000_000;
+/// This timeout is an argument in AESM request protobufs.
+///
+/// This value should be used for requests that might need interaction with
+/// remote servers, such as provisioning EPID.
+const REMOTE_AESM_TIMEOUT_US: u32 = 30_000_000;
+
+#[derive(Debug)]
+pub struct AesmClient {
+    tcp_stream: TcpStream,
+}
+
+impl AesmClient {
+    pub fn new(tcp_stream: TcpStream) -> Self {
+        AesmClient{ tcp_stream }
+    }
+
+    fn open_socket(&self) -> Result<TcpStream> {
+        let sock = self.tcp_stream.try_clone()?;
+        let _ = sock.set_write_timeout(Some(Duration::from_micros(LOCAL_AESM_TIMEOUT_US as _)))?;
```

Comment:
I've kept set_write_timeout and set_read_timeout, but I remember that they don't function correctly. 
Should I remove them?

Reply:
You can comment them out with a `FIXME` pointing to #31